### ISSUE TITLE
Twitter character counting

### ISSUE THREAD

**Issue by tiotrom** [friendica/friendica#9547]
I suggest that when you choose to also post on Twitter from Friendica, to add a character counting to the Friendica composer so the users know when they exceed the X character limit Twitter has.

Should work like this: if users have enabled to auto post on Twitter, when they compose Friendica posts and the text limit exceeds X characters, the composer should add a warning saying this exceeds Twitter character limit. 

If users are connected with their Twitter, and compose a Friendica post, but after that they enable to post on twitter in the second tab of the composing popup window, it should display the warning message when they do that.

Lastly, when you comment/reply to a Twitter post from Friendica it should, again warn about the character limit ;)

These could be helpful!

**Comment by MrPetovan** [comment]
The catch with character counting is that Twitter doesn't have BBCodes and replaces every URL with a fixed-length "shortened" one, while images are handled differently as well. So except when you are only entering plain text, we have no reliable way of knowing when exactly your post will exceed Twitter's character limit. Any BBCode and the count is off. Any image and the count is off. Any URL and the count will be off.

**Comment by tiotrom** [comment]
Ah I see....

> fixed-length "shortened" one,

Hm...so can't you take that fixed-lenght into account? Say you detect a url in the post, and consider that as a "fix" lenght.

**Comment by MrPetovan** [comment]
Yes, but It would only be applicable to Twitter. If we had another connector to another service with limited characters and custom rules for shortening URLs, then it wouldn't be applicable anymore. I'd rather have an official Twitter counter script than trying to fiddle with the basic one we have knowing that it won't be applicable to any other connector. So far I haven't found one.

**Comment by MrPetovan** [comment]
Never mind, I found it: https://github.com/twitter/twitter-text

So we could start giving an accurate character count for Twitter when the addon is enabled. We just need to add a Javascript hook to the current character counter.

**Comment by tiotrom** [comment]
Oh awesome! :)

**Comment by MrPetovan** [comment]
It still doesn't completely solve the BBCode issue, but at least it would be handled more accurately than now.

**Comment by tiotrom** [comment]
A suggestion: in case there are multiple connectors for multiple other networks like twitter, each with its own character limit, then you can count the characters and bellow the composer or comment box, showcase:

Twitter: X/Y
Other network: X/Z
Another one: X/N

Where X is the number of characters you counted, and Y, Z, and N, are the limited characters for those social platforms. A way for the user to know if it exceeded the character limits for any of the connectors.

You know what I mean?

**Comment by MrPetovan** [comment]
I was thinking of having a simpler display whether the current message would be fine for the according connector or if it would be truncated. So it would look something like this: `350 [Wordpress ✔] [Twitter ✂]`

**Comment by ghost** [comment]
WordPress doesn't have a character limit, does it?

**Comment by annando** [comment]
@MrPetovan it would be no problem to create some endpoint that could be called via JS. This could receive the content via `POST` and could return for which networks the content is suitable.

**Comment by MrPetovan** [comment]
> WordPress doesn't have a character limit, does it?

No but it was an example. I don't have any other example in mind of a connector enforcing a character limit.
